### ISSUE TITLE
code confusion and inconsistent reproduction results

### ISSUE THREAD

**Issue by dueToLife** [minghchen/CARL_code#3]
Congratulations, you've done a fantasitic job. I try to reproduce your work, but find some problems.
**First**, i noticed that  the parameter DATA.SAMPLING_REGION in *_config.yml refer to alpha variant in your paper., and `block_size` in dataset class's `sample_frame` function means sample size like [Pouring](https://github.com/minghchen/CARL_code/blob/c574312969c35963fc0050a5b5dbd3e4aa590318/datasets/pouring.py#L119). Should it be the production of `expand_ratio` and `num_frames` rather than `expand_ratio` and `seq_len` like 
```python
def sample_frames(self, seq_len, num_frames, pre_steps=None):
        ...
        elif sampling_strategy == 'time_augment':
            num_valid = min(seq_len, num_frames)
            expand_ratio = np.random.uniform(low=1.0, high=self.cfg.DATA.SAMPLING_REGION) if self.cfg.DATA.SAMPLING_REGION>1 else 1.0
            # block_size = math.ceil(expand_ratio*seq_len)
            block_size = math.ceil(expand_ratio*num_frames)
```
**Second**, i run training process in Pouring Dataset but only got 90.3% classification accuracy. It's less than 93.73% reported in your paper. Can you give some advice ? I use batch_size = 4, embedde_model.num_layers = 3, other parameter is the same with [scl_transformer_config](https://github.com/minghchen/CARL_code/blob/master/configs/scl_transformer_config.yml) in your repo.

**Comment by minghchen** [comment]
You do read the code and the paper very carefully! This is true that 'block_size' should be `math.ceil(expand_ratio*num_frames)` according to our paper. However, I found that setting 'block_size'  as `math.ceil(expand_ratio*seq_len)` would produce a little higher performance than 'math.ceil(expand_ratio*num_frames)'.

This means we do not perform temporally cropping (before sampling) for Pouring datasets, but we perform temporally padding. The reason why this works for Pouring is that we keep the development of the whole video when not cropping (the training frames more likely contain all five phases of the pouring action), and this is consistent with the test video (the frames of testing videos contain all five phases of the pouring action). We have discussed this (padding instead of cropping) for Pouring in our first version of the paper, but we omitted it at revision for brevity.

Maybe, you should try the exact same config where batch_size = 1 and embedde_model.num_layers = 2 to reproduce.

**Comment by dueToLife** [comment]
THX! That is true when all phases are included, the performance will be better. I am tring to use your method on dataset of our research field to do a pre-training process, and i wonder how do you determine the num of crop frame in each views, namely T in your paper? Is it determined by the different average frame length of different dataset? Thanks again for your answer about temporal crop and temporal padding, maybe i should only use temporal padding in our dataset which is a little similar like Pouring and PennAction.  :)

**Comment by minghchen** [comment]
It is true that we determine T according to the average frame length of training videos. Theoretically, if the average frame length is N, then T=N/2 will produce most frame combinations (simple combinatorial mathematics). We think more frame combinations will have stronger data augmentation. So we approximately chose T=N/2.
